### ISSUE TITLE
Website for Gorgonia

### ISSUE THREAD

**Issue by chewxy** [gorgonia/gorgonia.github.io#12]
Gorgonia needs a website, I've been told. There are many features people don't know about, and they should be documented. Here's a rough framework in mind

* [ ] Basic Section
    * [ ] Fundamentals: Computation Graphs
    * [ ] Fundamentals: Tensors
    * [ ] Fundamentals: Backpropagation and How It Relates to Differentiation
    * [ ] Fundamentals: Stochastic Gradient Descent (VanillaSGD and RMSProp)
    * [ ] Examples
        * [x] Logistic Regression
        * [x] Stacked Autoencoder
        * [x] RNNs
        * [ ] LSTM
* [ ] Advanced section
    * [ ] Different execution modes (`LispMachine` vs `TapeMachine`)
    * [ ] Optimizing the computation graph
    * [ ] Using MKL/CUDA
    * [ ] Exploring the various SGD solvers provided in Gorgonia: ADAM, AdaGrad, etc
    * [ ] Extending Gorgonia
        * [ ] New Ops
        * [ ] New Solvers
    * [ ] Debugging your neural network
* [ ] Behind the Scenes
   * [ ] The design of Tensors
    * [ ] Design Decisions and Rationales

**Comment by ouwe-knutselaar** [comment]
I've been looking at the website but it is still very hard to understand as a beginner what you are building.
After following the tutorials, I'm not able to use Gorgonia. The level is still too high.
There should be a more detailed explanation about what is build and why certain functions/types/decisions are made.
